src/main/java/uk/ac/ebi/eva/evaseqcol/datasource/NCBIAssemblySequenceDataSource.java

```diff
@@ -43,6 +44,17 @@ public NCBIAssemblySequenceDataSource(NCBIBrowserFactory factory,
         this.readerFactory = readerFactory;
     }
 
+    public Optional<AssemblySequenceEntity> getAssemblySequencesByAccession(String insdcAccession, String fastFileContent) throws IOException {
+        AssemblySequenceEntity assemblySequenceEntity;
+        try (InputStream stream = new ByteArrayInputStream(fastFileContent.getBytes())) {
+            NCBIAssemblySequenceReader reader = readerFactory.build(stream, insdcAccession);
```

Comment:
We'll probably have to refactor this since `NCBIAssemblySequenceReader` is actually pretty generic and does not need to be attached to 'NCBI' prefix